tests/unit/api/apiUtils/versioning.js

```diff
@@ -99,15 +98,14 @@ describe('versioning helpers', () => {
                 versioningEnabledExpectedRes: {
                     options: {
                         versioning: true,
-                        nullVersionId: 'vnull',
-                        nullUploadId: 'fooUploadId',
+                        extraMD: {
```

Comment:
I'm not sure I see the point of the `extraMD` nesting.

Reply:
The idea is to try to clarify the contract between `processVersioningState` and its callers, by explicitly grouping into an object the attributes should be added to the object metadata. The rest of parameters passed in `options` are specific options to the backend and/or to the calling code. What makes it still confusing in the current state is that the contract to the service call to actually store the object like e.g. `createAndStoreObject` takes params (passed via `metadataStoreParams` usually) which is also a huge flat object that mixes attributes to add, backend options and others. I did not go as far as to refactor how parameters are passed to the service (`metadataStoreParams`) but ideally this would also be clarified the same eventually, e.g. by grouping params by family, at least here it is supposed to make it clearer at this function's level even if the same attributes are passed in a flat object to the service helper thereafter.

Also, I changed the `storeParams` return value in that function, not specific enough as it specifically means to store a null version key from the master md in practice, into a `nullVersionId` returned value, which the caller takes as the hint that it should copy the master to this null version key (granted that it's not super specific either, I could call this `nullVersionIdToStore` maybe but it's getting a bit long of a name!)

Reply:
Moreover, the upcoming update to support null keys will add a few more fields to the response, so it looks better to start tidying things up a bit.